### PR TITLE
[SPARK-27961][SQL] DataSourceV2Relation should not have refresh method

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

The newly added `Refresh` method in PR #24401 prevented the work of moving DataSourceV2Relation into catalyst. It calls `case table: FileTable => table.fileIndex.refresh()` while `FileTable` belongs to sql/core.

More importantly, Ryan Blue pointed out DataSourceV2Relation is immutable by design, it should not have refresh method.

## How was this patch tested?

Unit test

### CODE REVIEW

**Comment by gengliangwang** [comment]
@rdblue @jzhuge @gatorsmile

**Comment by rdblue** [comment]
@gengliangwang, the catalog interface has a method to invalidate a source table. Do you want to implement `REFRESH TABLE` for v2 tables as a follow-up?

**Comment by rdblue** [comment]
+1 from me if tests pass. (I saw the comment about the Paruqet -> ORC fix on the previous PR.)

**Comment by gengliangwang** [comment]
> @gengliangwang, the catalog interface has a method to invalidate a source table. Do you want to implement REFRESH TABLE for v2 tables as a follow-up?

Yes, I will create a follow-up. This one is to unblock the work of moving DataSourceV2Relation into catalyst.

**Comment by dongjoon-hyun** [comment]
@gengliangwang . Could you update the PR description a little bit? `me` is you or John?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Logically, this reverts `[SPARK-27504][SQL] File source V2: support refreshing metadata cache`.

I have two questions.
- Are we going to make https://issues.apache.org/jira/browse/SPARK-27504 as `Invalid`?
- Why not reverting the invalid SPARK-27504 cleanly?

cc @gatorsmile

File: sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ abstract class MetadataCacheSuite extends QueryTest with SharedSQLContext {
   test("SPARK-16336,SPARK-27504 Suggest doing table refresh " +
     "when encountering FileNotFoundException") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala]: The test case name `Suggest doing table refresh` is not valid anymore DSv2 after this PR.

**Comment by gengliangwang** [comment]
@dongjoon-hyun I think we can partially remove it. Some of the works are still valuable:
1. Showing warning message in file source v2 if the underlying files have been updated 
2. The test suite is reserved.  Otherwise, we can see test failure in the Parquet V2 migration PR.

**Review by jzhuge — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala [hunk 2]
```diff
@@ -57,13 +56,20 @@ abstract class MetadataCacheSuite extends QueryTest with SharedSQLContext {
...
   }
+}
+
+class MetadataCacheV1Suite extends MetadataCacheSuite {
+  override protected def sparkConf: SparkConf =
+    super
+      .sparkConf
+      .set(SQLConf.USE_V1_SOURCE_READER_LIST, "orc")
 
   test("SPARK-16337,SPARK-27504 temporary view refresh") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala]: `SPARK-27504` had better be removed like [this](https://github.com/apache/spark/pull/24815/files#diff-0667b59236ca014a47b3fc20b6ea820eR41)?

**Comment by gengliangwang** [comment]
retest this please.

**Comment by gengliangwang** [comment]
@dongjoon-hyun Any other concerns on this PR?

**Comment by dongjoon-hyun** [comment]
Sorry for the delay, @gengliangwang . Yes, for now, this might be the less noise path because SPARK-27504 was merged long time ago.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
cc @gatorsmile since this PR has a history and the context.

**Comment by jzhuge** [comment]
Thanks @gengliangwang @rdblue @dongjoon-hyun for the quick resolution!

**Pull request closed without merging**